File: sudoku-auth/auth-api/src/main/java/com/zumin/sudoku/auth/feign/AuthFeign.java
package com.zumin.sudoku.auth.feign;


import com.zumin.sudoku.auth.pojo.dto.OAuth2TokenDTO;
import com.zumin.sudoku.common.core.result.CommonResult;
import java.util.Map;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient("sudoku-auth")
public interface AuthFeign {

  @PostMapping(value = "/oauth/token")
  CommonResult<OAuth2TokenDTO> postAccessToken(@RequestParam Map<String, String> parameters);

  @GetMapping("/captcha/check")
  CommonResult<?> checkCaptcha(@RequestParam("uuid") String uuid, @RequestParam("code") String code);
}

File: sudoku-auth/auth-api/src/main/java/com/zumin/sudoku/auth/AuthFeign.kt
package com.zumin.sudoku.auth

import com.zumin.sudoku.common.core.CommonResult
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(name = "sudoku-auth", contextId = "captcha")
interface CaptchaFeign {
  @GetMapping("/captcha/check")
  fun checkCaptcha(@RequestParam("uuid") uuid: String, @RequestParam("code") code: String): CommonResult<*>
}

@FeignClient(name = "sudoku-auth", contextId = "oauth")
interface OAuthFeign {
  @PostMapping("/oauth/token")
  fun postAccessToken(@RequestParam parameters: Map<String, String>): CommonResult<OAuth2TokenDTO>
}